src/H5FDmpio.c

```diff
@@ -2716,6 +2733,924 @@ H5FD__mpio_write_vector(H5FD_t *_file, hid_t H5_ATTR_UNUSED dxpl_id, uint32_t co
+#else
+            if (MPI_SUCCESS != (mpi_code = MPI_Get_elements(&mpi_stat, MPI_BYTE, &bytes_read))) {
+#endif
+                if (rank0_bcast && file->mpi_rank == 0) {
+                    /* If MPI_Get_elements(_x) fails for a rank 0 bcast strategy,
+                     * push an error, but continue to participate in the following
+                     * MPI_Bcast.
+                     */
+                    bytes_read = -1;
+                    HMPI_DONE_ERROR(FAIL, "MPI_Get_elements failed", mpi_code)
+                }
+                else
+                    HMPI_GOTO_ERROR(FAIL, "MPI_Get_elements failed", mpi_code)
+            }
+        } /* end if */
+
+        /* If the rank0-bcast feature was used, broadcast the # of bytes read to
+         * other ranks, which didn't perform any I/O.
+         */
+        /* NOTE: This could be optimized further to be combined with the broadcast
+         *          of the data.  (QAK - 2019/1/2)
+         */
+        if (rank0_bcast)
+#if H5_CHECK_MPI_VERSION(3, 0)
+            if (MPI_SUCCESS != MPI_Bcast(&bytes_read, 1, MPI_COUNT, 0, file->comm))
+#else
+            if (MPI_SUCCESS != MPI_Bcast(&bytes_read, 1, MPI_INT, 0, file->comm))
+#endif
+                HMPI_GOTO_ERROR(FAIL, "MPI_Bcast failed", 0)
+
+                /* Get the type's size */
+#if H5_CHECK_MPI_VERSION(3, 0)
+        if (MPI_SUCCESS != (mpi_code = MPI_Type_size_x(final_mtype, &type_size)))
+#else
+        if (MPI_SUCCESS != (mpi_code = MPI_Type_size(final_mtype, &type_size)))
+#endif
+            HMPI_GOTO_ERROR(FAIL, "MPI_Type_size failed", mpi_code)
+
+        /* Compute the actual number of bytes requested */
+        io_size = type_size * size_i;
+
+        /* Check for read failure */
+        if (bytes_read < 0 || bytes_read > io_size)
+            HGOTO_ERROR(H5E_IO, H5E_READERROR, FAIL, "file read failed")
+
+#ifdef H5FDmpio_DEBUG
+        if (H5FD_mpio_debug_r_flag)
+            fprintf(stderr, "%s: (%d) mpi_off = %ld  bytes_read = %lld  type = %s\n", __func__,
+                    file->mpi_rank, (long)mpi_off, (long long)bytes_read, H5FD__mem_t_to_str(type));
+#endif
+
+        /*
+         * This gives us zeroes beyond end of physical MPI file.
+         */
+        if ((n = (io_size - bytes_read)) > 0)
+            memset((char *)bufs[0] + bytes_read, 0, (size_t)n);
+
+    } /* end if */
+    else {
+#ifdef H5FDmpio_DEBUG
+        if (H5FD_mpio_debug_r_flag)
+            fprintf(stderr, "%s: (%d) doing MPI independent IO\n", __func__, file->mpi_rank);
+#endif
+        if (_file->base_addr > 0) {
+            /* Undo base address addition in internal routines before passing down to the mpio driver */
+            for (i = 0; i < count; i++) {
+                assert(offsets[i] >= _file->base_addr);
+                offsets[i] -= _file->base_addr;
+            }
+        }
+
+        if (H5FD_read_vector_from_selection(_file, type, count, mem_space_ids, file_space_ids, offsets,
```

Comment:
I think this should actually be H5FD_read_from_selection(). Same for write (I know my earlier email said vector)

Reply:
Done